content/api/cypress-api/custom-queries.md

```diff
@@ -0,0 +1,385 @@
+loaded before any test files are evaluated via an import statement in the
+[supportFile](/guides/core-concepts/writing-and-organizing-tests#Support-file).
+
+</Alert>
+
+Queries follow three main rules.
+
+1. Queries are _synchronous._ They do not return or await promises.
+2. Queries are _retriable._ Once you return the inner function, Cypress takes
+   control, handling retries on your behalf.
+3. Queries are _idempotent._ Once you return the inner function, Cypress will
+   invoke it repeatedly. Invoking the inner function multiple times must not
+   change the state of your application.
+
+With these rules, queries are simple to write and extremely powerful. They are
+the building blocks on which Cypress' API is built.
+
+## Syntax
+
+```javascript
+Cypress.Commands.addQuery(name, callbackFn)
+```
+
+### Usage
+
+**<Icon name="check-circle" color="green"></Icon> Correct Usage**
+
+```javascript
+Cypress.Commands.addQuery('getById', function (id) {
+  return (subject) => newSubject
+})
+```
+
+### Arguments
+
+**<Icon name="angle-right"></Icon> name** **_(String)_**
+
+The name of the query you're adding.
+
+**<Icon name="angle-right"></Icon> callbackFn** **_(Function)_**
+
+Pass a function that receives the arguments passed to the query.
+
+This outer function is invoked once. It should return a function that takes a
+subject and returns a new subject; this inner function might be called multiple
+times.
+
+## Examples
+
+### `.focused()`
+
+The callback function can be thought of as two separate parts. The _outer
+function_, which is invoked once, where you perform setup and state management,
+and the _query function_, which might be called repeatedly.
+
+Let's look at an example. This is actual Cypress code - how `.focused()` is
+implemented internally. The only thing omitted here for simplicity is the
+TypeScript definitions.
+
+```javascript
+Commands.addQuery('focused', function focused(options = {}) {
+  const log = options.log !== false && Cypress.log({ timeout: options.timeout })
+
+  this.set('timeout', options.timeout)
+
+  return () => {
+    let $el = cy.getFocused()
+
+    log &&
+      cy.state('current') === this &&
+      log.set({
+        $el,
+        consoleProps: () => {
+          return {
+            Yielded: $el?.length ? $dom.getElements($el) : '--nothing--',
+            Elements: $el != null ? $el.length : 0,
+          }
+        },
+      })
+
+    if (!$el) {
+      $el = $dom.wrap(null)
+      $el.selector = 'focused'
+    }
+
+    return $el
+  }
+})
+```
+
+#### The outer function
+
+The outer function is called once each time test uses the query. It performs
+setup and state management:
+
+```javascript
+function focused(options = {}) {
+  const log = options.log !== false && Cypress.log({ timeout: options.timeout })
+
+  this.set('timeout', options.timeout)
+
+  return () => {} // Inner function
+}
+```
+
+Piece by piece, then.
+
+```javascript
+function focused(options = {}) {}
+```
```

Comment:
Not sure where this is tied in

Reply:
I've added "..." to indicate that I'm omitting the function body. 

But this is just "starting at the top." It's the first line of the outer function. I wanted to talk about it, so I'm repeating it, same as I do for each other line.